grpc-proto/proto/schema.proto

```diff
@@ -141,3 +141,20 @@ message SchemaRead {
 message AuthorizeSchemaReadsResponse {
   repeated bool authorized = 1;
 }
+
+// The arguments to a GetSupportedFeatures call.
+// Intentionally empty, there are no arguments at this time but this is intended for future
+// extensibility.
+message SupportedFeaturesRequest {}
+
+// The response to a GetSupportedFeatures call.
+message SupportedFeaturesResponse {
+  // Whether the persistence backend supports regular Cassandra secondary indexes.
+  bool secondary_indexes = 1;
+
+  // Whether the persistence backend supports Storage Attached Indexes.
+  bool sai = 2;
+
+  // Whether the persistence backend supports logged CQL batches.
+  bool logged_batches = 3;
+}
```

Comment:
My idea was to solve this with configuration on the Docs API.. I want to avoid having to make this call.. Meaning I will have something like:

```
stargate.data-store-support.secondaryIndexes=
stargate.data-store-support.sai=
stargate.data-store-support.logged-batch=
```

Then I would create profiles for each supported data store, for example, `application-DSE.yaml` which would contain props for the DSE. Then when starting the microservice you can select the profile based on the underlying data store.

Not saying that this is not good, just explaining the option I had in mind. This makes less configuration, but with that extra call.. Not sure if this makes testing harder, but yea I am fine with both options..

Reply:
> My idea was to solve this with configuration on the Docs API.

The downside is that this relies entirely on the user getting it right... If there is a configuration error, the service will try to use the wrong features and get errors.

> I want to avoid having to make this call.

Yeah that's a valid concern.
In the GraphQL API it doesn't really matter because the call is not on a hot path.
But if needed I could memoize the result at the `StargateBridgeClientFactory` level, so that the gRPC call is only done once (the response is always the same for a given persistence implementation).

Reply:
This is all-around problematic because conceptually Service is asking for Persistence-side feature and ideally it would ask and not be statically configured. But I do understand that for practical reasons it may be necessary for Service to configure such settings so as not to have to connect to Persistence.

Perhaps we cannot unify all such introspection behind same API; so maybe changes here should be configured wrt how well they work for GraphQL use case. And @ivansenic's settings may need to be handled differently.
I am not sure this is the way to go but wanted to mentioned that different use cases sometimes require different solutions.

Reply:
> we can call it once on the app startup and consider that it's fixed.. But this is not possible with the current routing approach, as you need to do it in the request context.

We can do it for the first request that asks for it, and then cache the result.

Reply:
True.